rotkehlchen/rotkehlchen.py

```diff
@@ -173,35 +195,108 @@ def __init__(self, args: argparse.Namespace) -> None:
+
+        # Acquire everything what's possible at this point. Other locks will be acquired
+        # further down the line.
+        for lock in all_locks:
+            if lock.locked() is False:
+                stack.enter_context(lock)
+                acquired_by_this_method[lock] = True
+
         for address in addresses:
+            # Check all api task greenlets
             account_tuple = (address, blockchain.to_chain_id())
             for greenlet in self.api_task_greenlets:
                 is_evm_tx_greenlet = (
                     len(greenlet.args) >= 1 and
-                    isinstance(greenlet.args[0], MethodType) and
-                    greenlet.args[0].__func__.__qualname__ == 'RestAPI._get_evm_transactions'
+                    isinstance(greenlet.args[0], MethodType)
                 )
                 if (
                         is_evm_tx_greenlet and
+                        greenlet.args[0].__func__.__qualname__ == 'RestAPI._get_evm_transactions' and  # noqa: E501
                         greenlet.kwargs['only_cache'] is False and
                         account_tuple in greenlet.kwargs['filter_query'].accounts
                 ):
+                    kill_greenlet_and_acquire_locks(
+                        stack=stack,
+                        acquired_by_this_method=acquired_by_this_method,
+                        greenlet=greenlet,
+                        locks=[evm_manager.transactions.address_tx_locks[address]],
+                    )
+                elif (
+                    is_evm_tx_greenlet and
+                    greenlet.args[0].__func__.__qualname__ == 'RestAPI._decode_pending_evm_transactions'  # noqa: E501
+                ):
+                    kill_greenlet_and_acquire_locks(
+                        stack=stack,
+                        acquired_by_this_method=acquired_by_this_method,
+                        greenlet=greenlet,
+                        locks=[
+                            evm_manager.transactions_decoder.undecoded_tx_query_lock,
+                            evm_manager.transactions.missing_receipts_lock,
+                        ],
+                    )
 
-                    greenlet.kill()
+            # Check related periodic tasks greenlets
+            for greenlet in self.task_manager.running_greenlets.get(self.task_manager._maybe_query_evm_transactions, []):  # noqa: E501
+                address = greenlet.kwargs.get('address')
+                if address is None or address in addresses:
+                    kill_greenlet_and_acquire_locks(
+                        stack=stack,
+                        acquired_by_this_method=acquired_by_this_method,
+                        greenlet=greenlet,
+                        locks=[evm_manager.transactions.address_tx_locks[address]],
+                    )
+
+            for greenlet in self.task_manager.running_greenlets.get(self.task_manager._maybe_decode_evm_transactions, []):  # noqa: E501
```

Comment:
This is not needed. 

1. Why is it that important to kill a transaction decodign task?
2. This won't kill the transaction decoding task.Since this will kill just the scheduling event. This is where the scheduling event will actually spawn the decoding task: https://github.com/rotki/rotki/blob/97ec324c0216e55cb8aa1a69d0610ff1030a4d62/rotkehlchen/tasks/manager.py#L476